### ISSUE TITLE
Investigate GNU/Posix MacOS distribution issues

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#1115]
# Task Description

We are seeing the following issue with MacOS builds:
```
ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.1.1:single (default) on project strongbox-distribution: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:3.1.1:single failed: group id ‘960648751’ is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit -> [Help 1]
```

@forbob contributed #1008, which I had to revert (as part of #1068), because it was causing the `strongbox-web-core-1.0-SNAPSHOT-spring-boot.jar` to have it’s name truncated in the `strongbox-distribution` to something like `strongbox-web-core-1.0-SNAPSHOT`, inside the `strongbox-distribution` ‘s `tar.gz`.

We need a permanent solution that works both on Linux, Windows and MacOS, which doesn’t break the `strongbox-distribution*tar.gz` (I believe it’s a bug in the `maven-assembly-plugin`.

Relates to #1007, #1008, #1068.

# Acceptance Test

* [ ] The `tar.gz` file produced by the `strongbox-distribution` contains proper file names which aren't truncated. for example, the `strongbox-1.0-SNAPSHOT/lib/` should contain `strongbox-web-core-1.0-SNAPSHOT-spring-boot.jar`, not a truncated file name like `strongbox-web-core-1.0-SNAPSHOT-spring`).
* [ ] Building the code and running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` and `tar.gz` still works.
* [ ] Running the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

**Comment by cbrianhill** [comment]
I reproduced this problem.  On MacOS, users with very large UIDs / GIDs are not able to create the tar archives, the `mvn install` failing with the message reported above.  Users with smaller UID / GID are able to build successfully.

I discovered much more about the issue where filenames were truncated.  As reported above, configuring the maven-assembly-plugin to use POSIX tar file format extensions allows large UID / GID to be recorded in the tar file.  The file truncation reported above occurred when the tar file was opened in `mc` on Linux.  The problem is actually a manifestation of this issue in `mc`:

https://midnight-commander.org/ticket/1952

Midnight Commander does not support POSIX tar file extensions, resulting in the files being displayed with truncated names.  We would generally prefer to keep the tar file extensions using the GNU format, because it's so widely supported.  The only option to fix the UID / GID issue would be to specify a specific UID / GID when creating the tar archive.  The GNU tar command line utility allows us to do this using the `--owner` and `--group` options.  However, this is not supported by Plexus Archiver, which is used by maven assembly plugin to create the tar archives.  This was previously reported under the following issue, but I wasn't able to find an issue logged against Plexus Archiver:

https://issues.apache.org/jira/browse/MASSEMBLY-772

This leaves us needing to choose between having to build with a sufficiently small UID / GID user, and distributing tar files that are not supported by `mc`.  I suggest we close this issue, with the workaround that MacOS users would need to either create a local user (large UID / GID are typical artifacts of a Mac joined to a Windows domain or some other managed account service) with a small UID / GID to build, or build using a Linux VM.
